pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -328,6 +331,47 @@ func buildNetworkTopologyMetadata(deviceID string, store *metadata.Store) []meta
 	return links
 }
 
+func resolveLocalInterface(deviceID string, interfaceIndexByIDType map[string]map[string]int32, localInterfaceIDType string, localInterfaceID string) (string, string) {
+	if localInterfaceID == "" {
+		return localInterfaceIDType, localInterfaceID
+	}
+
+	var typesToTry []string
+	if localInterfaceIDType == "" {
+		// "smart resolution" by multiple types when localInterfaceIDType is not provided (which is often the case)
+		// CAVEAT: in case the smart resolution returns false positives, the solution is to configure the device to provide a proper localInterfaceIDType
+		typesToTry = []string{"interface_name", "mac_address", "interface_name", "interface_alias", "interface_index"}
```

Comment:
```suggestion
		typesToTry = []string{"mac_address", "interface_name", "interface_alias", "interface_index"}
```
Seems like `interface_name` was duplicated, I assume `mac_address` should be the one tested first? 
Can we also add a comment about why we are using this order? I'm curious for instance why `interface_name` is tried before `interface_index`?

Reply:
@FlorianVeaux Good catch.

Fixed here: https://github.com/DataDog/datadog-agent/pull/14991/commits/a5c252c8e0914e32463db47fef1e06020823f93a

And added comment here: https://github.com/DataDog/datadog-agent/pull/14991/commits/32f7a915be95d81c81c7378e04c7de83a9048810

> 		// The order of `typesToTry` has been arbitrary define (not sure if there is an order that can lead to lower false positive).


WDYT? Do you see an order that can reduces the number of false positives?

What we can possibly do, is to analyse all IDs and if there is statistically a type that appears in majority e.g. `mac_address`, we can prioritise this one, but I'm not sure it's worth doing since it can lead to more complex code with benefits hard to determine.

I think if false positives are seen, the best solution is still:
> CAVEAT: in case the smart resolution returns false positives, the solution is to configure the device to provide a proper localInterfaceIDType

Reply:
Based on SNMP Walks we have, all 3 mac_address, interface_name, interface_index appear quite often in similar frequency. (didn't found example for interface_alias).

Reply:
I think that I was more thinking about the risk based on the type itself.
For instance, there is little risk to mix a mac_address with something else because of it's format. `interface_name` on the other hand might be set to something like "12" for whatever reason which could collide with the ifIndex.
I guess that's a good reason for trying ifIndex last maybe?

Reply:
@FlorianVeaux I made some quick stats using the snmp walks we have, here are the results

```
1/ (IF INDEX) ID contain only digits
1\.0\.8802\.1\.1\.2\.1\.3\.7\.1\.3.*(STRING : "|\|)\d+"?$
264 matches across 9 files

2/ (IF NAME)ID starts with 1/ or starts with a letter
1\.0\.8802\.1\.1\.2\.1\.3\.7\.1\.3.*(STRING : "|\|)(1\/\d+|[a-zA-Z].*)"?$
1142 matches across 23 files

3/ (MAC ADDRESS)
1\.0\.8802\.1\.1\.2\.1\.3\.7\.1\.3.*(STRING : "|\|)([a-zA-Z0-9]{12})"?$
207 matches across 9 files
```

Based on those stats, there is definitely more cases of IF NAME compared to IF INDEX and MAC ADDRESS.

So it can make sense to prioritise IF NAME vs IF INDEX.

For MAC ADDRESS, I think we should still have it first, since it's unlikely that something else will be matched with a MAC ADDRESS.

--- 

This said, the only safe way is to provide explicitly the `localInterfaceIDType` type.